src/realm/object-store/shared_realm.cpp

```diff
@@ -138,17 +138,17 @@ std::shared_ptr<DB>& Realm::Internal::get_db(Realm& realm)
     return realm.m_coordinator->m_db;
 }
 
-void Realm::Internal::begin_read(Realm& realm, VersionID version_id)
+void Realm::Internal::begin_read(Realm& realm, VersionID version_id, bool is_schema_additive)
 {
-    realm.begin_read(version_id);
+    realm.begin_read(version_id, is_schema_additive);
 }
 
-void Realm::begin_read(VersionID version_id)
+void Realm::begin_read(VersionID version_id, bool is_schema_additive)
```

Comment:
Is this extra bool really necessary at such a high level? The Realm instance can tell if it is additive mode or not because it has the config right? Can't we just use `read_schema_from_group_if_needed(m_config.is_schema_additive())` below?

Reply:
I think the name of the flag is misleading, it is used for fetching the schema from the version passed rather than loading the latest (in case the schema is additive)

Reply:
Just to wrap this up, the problem is that we need in some cases for additive realms to avoid returning the OS schema if the version is passed. I am not sure if there is another way to do this, rather than adding this boolean flag, adding a new API or a new enum param. 
Alternatively, we can accept that this test: 
https://github.com/realm/realm-core/blob/a0c5e41a7f0f6cf115b46b1367ce684ce521b38a/test/object-store/realm.cpp#L358
will not show any difference between `old_version` and `new_version` if the schema is additive.